src/main/java/hudson/tasks/Mailer.java

```diff
@@ -502,29 +518,60 @@ public void setHudsonUrl(String hudsonUrl) {
          *      Use {@link JenkinsLocationConfiguration} instead
          * @param adminAddress Jenkins administrator mail address to set
          */
+        @Deprecated
         public void setAdminAddress(String adminAddress) {
             getJenkinsLocationConfiguration().setAdminAddress(adminAddress);
         }
 
+        @DataBoundSetter
         public void setSmtpHost(String smtpHost) {
-            this.smtpHost = smtpHost;
+            this.smtpHost = nullify(smtpHost);
+            save();
         }
 
+        @DataBoundSetter
         public void setUseSsl(boolean useSsl) {
             this.useSsl = useSsl;
+            save();
         }
 
+        @DataBoundSetter
         public void setSmtpPort(String smtpPort) {
             this.smtpPort = smtpPort;
+            save();
         }
-        
-        public void setCharset(String chaset) {
-            this.charset = chaset;
+
+        @DataBoundSetter
+        public void setCharset(String charset) {
+            if (charset == null || charset.length() == 0) {
+                charset = "UTF-8";
+            }
+            this.charset = charset;
+            save();
+        }
+
+        @DataBoundSetter
+        public void setAuthentication(@CheckForNull SMTPAuthentication authentication) {
+            this.authentication = authentication;
+            save();
         }
 
+        @CheckForNull
+        public SMTPAuthentication getAuthentication() {
+            return authentication;
+        }
+
+        /**
+         * @deprecated as of 1.21
+         *      Use {@link #authentication}
+         */
+        @Deprecated
         public void setSmtpAuth(String userName, String password) {
-            this.smtpAuthUsername = userName;
-            this.smtpAuthPassword = Secret.fromString(password);
+            if (userName == null && password == null) {
```

Comment:
This would allow one of them being null. Is that expected?

Reply:
it's the same behaviour than in the current version